### ISSUE TITLE
BUG: Unhandled error in plugin series: (sqlite3.OperationalError) disk I/O error

### ISSUE THREAD

**Issue by lamski** [Flexget/Flexget#2355]
Started crashing yesterday on flexget 2.20.11 so I upgraded to 2.20.15. The same crash with SQLite still occurs.

I'm not really sure if this is related, but I noticed the crash started happening when I added this show to the series plugin:
```
        - Arrested Development:
            begin: S05E09
```

Update: removed the show above from the config and the error still occurs.

#### Log:
<details>
 <summary>(click to expand)</summary>

```
2019-03-27 02:35 INFO     manager                       Daemonizing...
2019-03-27 02:35 INFO     manager                       Daemonize complete. New PID: 18752
2019-03-27 02:35 INFO     scheduler                     Starting scheduler
2019-03-27 03:00 CRITICAL task          TVRSS           BUG: Unhandled error in plugin series: (sqlite3.OperationalError) disk I/O error [SQL: 'SELECT series_episodes.id AS series_episodes_id, series_episodes.identifier AS series_episodes_identifier, series_episodes.season AS series_episodes_season, series_episodes.number AS series_episodes_number, series_episodes.identified_by AS series_episodes_identified_by, series_episodes.series_id AS series_episodes_series_id \nFROM series_episodes JOIN episode_releases ON series_episodes.id = episode_releases.episode_id JOIN series ON series.id = series_episodes.series_id \nWHERE series.id = ? AND episode_releases.downloaded = 1 AND series_episodes.identified_by = ? ORDER BY series_episodes.season DESC, series_episodes.number DESC\n LIMIT ? OFFSET ?'] [parameters: (20, 'ep', 1, 0)] (Background on this error at: http://sqlalche.me/e/e3q8)
2019-03-27 03:00 CRITICAL manager       TVRSS           An unexpected crash has occurred. Writing crash report to /shares/conf/flexget/crash_report.2019.03.27.030028905440.log. Please verify you are running the latest version of flexget by using "flexget -V" from CLI or by using version_checker plugin at http://flexget.com/wiki/Plugins/version_checker. You are currently using version 2.20.15
2019-03-27 03:00 WARNING  task          TVRSS           Aborting task (plugin: series)
2019-03-27 03:27 INFO     remember_rej  thepiratebay    Remembering rejection of `This.Is.Us.S03E14.HDTV.x264-SVA%5Bettv%5D`
2019-03-27 03:27 INFO     remember_rej  thepiratebay    Remembering rejection of `The.Rookie.S01E15.HDTV.x264-KILLERS%5Bettv%5D`
2019-03-27 03:27 INFO     remember_rej  thepiratebay    Remembering rejection of `The.Flash.2014.S05E15.HDTV.x264-SVA%5Bettv%5D`
2019-03-27 03:27 CRITICAL task          discover_series BUG: Unhandled error in plugin discover: (sqlite3.OperationalError) disk I/O error [SQL: 'SELECT series_episodes.id AS series_episodes_id, series_episodes.identifier AS series_episodes_identifier, series_episodes.season AS series_episodes_season, series_episodes.number AS series_episodes_number, series_episodes.identified_by AS series_episodes_identified_by, series_episodes.series_id AS series_episodes_series_id \nFROM series_episodes JOIN episode_releases ON series_episodes.id = episode_releases.episode_id JOIN series ON series.id = series_episodes.series_id \nWHERE series.id = ? AND episode_releases.downloaded = 1 AND series_episodes.identified_by = ? ORDER BY series_episodes.season DESC, series_episodes.number DESC\n LIMIT ? OFFSET ?'] [parameters: (1, 'ep', 1, 0)] (Background on this error at: http://sqlalche.me/e/e3q8)
2019-03-27 03:27 CRITICAL manager       discover_series An unexpected crash has occurred. Writing crash report to /shares/conf/flexget/crash_report.2019.03.27.032729371818.log. Please verify you are running the latest version of flexget by using "flexget -V" from CLI or by using version_checker plugin at http://flexget.com/wiki/Plugins/version_checker. You are currently using version 2.20.15
2019-03-27 03:27 WARNING  task          discover_series Aborting task (plugin: discover)
2019-03-27 03:30 CRITICAL task          TVRSS           BUG: Unhandled error in plugin series: (sqlite3.OperationalError) disk I/O error [SQL: 'SELECT series_episodes.id AS series_episodes_id, series_episodes.identifier AS series_episodes_identifier, series_episodes.season AS series_episodes_season, series_episodes.number AS series_episodes_number, series_episodes.identified_by AS series_episodes_identified_by, series_episodes.series_id AS series_episodes_series_id \nFROM series_episodes JOIN episode_releases ON series_episodes.id = episode_releases.episode_id JOIN series ON series.id = series_episodes.series_id \nWHERE series.id = ? AND episode_releases.downloaded = 1 AND series_episodes.identified_by = ? ORDER BY series_episodes.season DESC, series_episodes.number DESC\n LIMIT ? OFFSET ?'] [parameters: (20, 'ep', 1, 0)] (Background on this error at: http://sqlalche.me/e/e3q8)
2019-03-27 03:30 CRITICAL manager       TVRSS           An unexpected crash has occurred. Writing crash report to /shares/conf/flexget/crash_report.2019.03.27.033027907731.log. Please verify you are running the latest version of flexget by using "flexget -V" from CLI or by using version_checker plugin at http://flexget.com/wiki/Plugins/version_checker. You are currently using version 2.20.15
2019-03-27 03:30 WARNING  task          TVRSS           Aborting task (plugin: series)
```
</details>

### Additional information:

- FlexGet version: 2.20.15
- Python version: 3.4.3
- Installation method: pip
- Using daemon (yes/no): yes
- OS and version: Ubuntu 14.04.6 LTS
- Link to crash log:
[crash_report.2019.03.27.030028905440.log](https://github.com/Flexget/Flexget/files/3010062/crash_report.2019.03.27.030028905440.log)
[crash_report.2019.03.27.032729371818.log](https://github.com/Flexget/Flexget/files/3010065/crash_report.2019.03.27.032729371818.log)

**Comment by lamski** [comment]
Turned on debug logging and uploaded the first set of flexget logs and crash report:
[flexget.log.debug.2355.zip](https://github.com/Flexget/Flexget/files/3015062/flexget.log.debug.2355.zip)
[crash_report.2019.03.28.033101423975.log](https://github.com/Flexget/Flexget/files/3015064/crash_report.2019.03.28.033101423975.log)

Should I backup and reset the sqlite DB?

**Comment by lamski** [comment]
Update: my db-config.sqlite DB is around 30 MB. I tried copying it to a different location and ubuntu's cp command fails with this message
```
cp: failed to extend <destination file> input/output error
```
The copied file would be around 22 MB only.

I was able to fully copy the DB file using the scp command.
I was able to load the DB on my IDE using the sqlite JDBC driver.
I then tried to run the SQL from some of the crash logs and substituting the parameters:
```
SELECT series_episodes.id AS series_episodes_id, series_episodes.identifier AS series_episodes_identifier, series_episodes.season AS series_episodes_season, series_episodes.number AS series_episodes_number, series_episodes.identified_by AS series_episodes_identified_by, series_episodes.series_id AS series_episodes_series_id 
FROM series_episodes JOIN episode_releases ON series_episodes.id = episode_releases.episode_id JOIN series ON series.id = series_episodes.series_id 
WHERE series.id = 20 AND episode_releases.downloaded = 1 AND series_episodes.identified_by = 'ep' ORDER BY series_episodes.season DESC, series_episodes.number DESC
LIMIT 1 OFFSET 0
```
I then get this error:
```
[Exception, Error code 11, SQLState null] [SQLITE_CORRUPT]  The database disk image is malformed (database disk image is malformed)
```

Since I already have several backups, I tried cleaning, vacuuming, and resetting the DB but they all fail with sqlite IO errors as the earlier crashes.

I finally deleted the corrupt db-config.sqlite file and restarted the daemon. This seems to have fixed the issue. New torrents are now being added to my transmission client.

**Comment by lamski** [comment]
I managed to recover most of the data using this guide: https://techblog.dorogin.com/sqliteexception-database-disk-image-is-malformed-77e59d547c50

During restore of the SQL dump file to a new sqlite DB, I used the error messages to know which lines to delete.

When I got rid of the SQL errors, I was puzzled because the DB remained empty even after successfully running the SQL. It turned out that the script had a rollback statement at the end. Removing that line will restore the data. Make sure you commit the changes for the data to be written to the DB file.
